pkg/controller/jobframework/setup_test.go

```diff
@@ -134,18 +159,48 @@ func TestSetupControllers(t *testing.T) {
 				t.Fatalf("Failed to setup manager: %v", err)
 			}
 
-			gotError := manager.setupControllers(mgr, logger, tc.opts...)
+			gotError := manager.setupControllers(ctx, mgr, logger, tc.opts...)
 			if diff := cmp.Diff(tc.wantError, gotError, cmpopts.EquateErrors()); len(diff) != 0 {
 				t.Errorf("Unexpected error from SetupControllers (-want,+got):\n%s", diff)
 			}
 
-			if diff := cmp.Diff(tc.wantEnabledIntegrations, manager.enabledIntegrations.SortedList()); len(diff) != 0 {
+			if len(tc.delayedGVKs) > 0 {
+				// Simulate the delayed addition of RayCluster
+				simulateDelayedIntegration(mgr, tc.delayedGVKs)
+				for _, gvk := range tc.delayedGVKs {
+					testDelayedIntegration(&manager, gvk.Group+"/"+strings.ToLower(gvk.Kind))
+				}
+			}
+
+			diff := cmp.Diff(tc.wantEnabledIntegrations, manager.getEnabledIntegrations().SortedList())
+			if len(diff) != 0 {
 				t.Errorf("Unexpected enabled integrations (-want,+got):\n%s", diff)
 			}
 		})
 	}
 }
```

Comment:
You can do something like this:

```golang
func TestRESTMapper struct {
  *DefaultRESTMapper
  lock sync.RWMutex
}

func (m *TestRESTMapper) RESTMapping(...) {
  m.lock.Lock()
  return m.DefaultRESTMapper.RESTMapping(...)
}
```

Reply:
That is genius, please see latest changes if it meets with expectations. Thanks for the help and guidance Aldo.